spec/lib/miq_expression/tag_spec.rb

```diff
@@ -2,19 +2,54 @@
 
 RSpec.describe MiqExpression::Tag do
   describe ".parse" do
-    it "with managed-field" do
-      field = "managed.location"
-      expect(described_class.parse(field)).to eq(described_class.new(nil, "/location"))
-    end
-
     it "with model.managed-in_field" do
       field = "Vm.managed-service_level"
-      expect(described_class.parse(field)).to eq(described_class.new(Vm, "/managed/service_level"))
+      expect(described_class.parse(field)).to eq(described_class.new(Vm, "/managed/service_level", []))
     end
 
     it "with model.last.managed-in_field" do
       field = "Vm.host.managed-environment"
-      expect(described_class.parse(field)).to eq(described_class.new(Vm, "/host"))
+      expect(described_class.parse(field)).to eq(described_class.new(Vm, "/managed/environment", ['host']))
```

Comment:
This is testing `.parse` but also relies on `eql?/==` being implemented correctly. Is it only there to test this behavior? I think a better way to test it might be (since you're changing this anyway):

```ruby
expect(described_class.parse(field)).to have_attributes(:model => Vm, :namespace => "/managed/environment", :associations => ["host"])
```

Then we could remove `eql?` etc, if it's only there to facilitate testing.